ranger.py

```diff
@@ -149,3 +177,74 @@ def __init__(self, level):
             ],
             base_feats=base_feats,
         )
+
+
+class BeastMasterRanger(Character):
+    def __init__(self, level):
+        self.magic_weapon = get_magic_weapon(level)
+        base_feats = []
+        maul = BeastMaul(base=2 + prof_bonus(level))
+        shortsword = Shortsword(bonus=self.magic_weapon)
+        rapier = Rapier(bonus = self.magic_weapon)
+        other_shortsword = Shortsword(bonus=self.magic_weapon, name="Offhand Shortsword", base="dex" if level >= 2 else 0)
+        scimitar = Scimitar(bonus=self.magic_weapon, base="dex" if level >= 2 else 0)
+        base_feats.append(EquipWeapon(maul))
+        base_feats.append(EquipWeapon(shortsword))
+        base_feats.append(EquipWeapon(rapier))
+        base_feats.append(EquipWeapon(scimitar))
+        base_feats.append(EquipWeapon(other_shortsword))
+        def attacks():
```

Comment:
I'm wondering if it may be simpler to just duplicate `RangerAction` and modify it to do what you want, with methods for things like "attack with bear" and the like. There's a bit of duplicate code here that may be reduced with that implementation